packages/kvstore/lib/index.ts

```diff
@@ -1,17 +1,42 @@
+import type { RedisClientType } from 'redis';
+import { createClient } from 'redis';
 import { InMemoryKVStore } from './InMemoryStore.js';
 import { RedisKVStore } from './RedisStore.js';
 
 export { InMemoryKVStore } from './InMemoryStore.js';
 export { RedisKVStore } from './RedisStore.js';
 export type { KVStore } from './KVStore.js';
+export { FeatureFlags } from './FeatureFlags.js';
+export type { RedisClientType } from 'redis';
 
+export const redisURL = process.env['NANGO_REDIS_URL'];
+let kvStore: RedisKVStore | InMemoryKVStore | undefined;
 export async function createKVStore() {
-    const url = process.env['NANGO_REDIS_URL'];
-    if (url) {
-        const store = new RedisKVStore(url);
-        await store.connect();
-        return store;
+    if (kvStore) {
+        return kvStore;
     }
 
-    return new InMemoryKVStore();
+    if (redisURL) {
+        kvStore = new RedisKVStore(redisURL);
+        await kvStore.connect();
+    } else {
+        kvStore = new InMemoryKVStore();
+    }
+    return kvStore;
+}
+
+let redisClient: RedisClientType | undefined;
+export function createRedisClient({ url = redisURL, cache = true }: { url?: string; cache?: boolean } = {}): RedisClientType {
```

Comment:
It is not super obvious what the cache parameter is actually doing unless you look at the body of the function.
What about instead having 2 functions:
- `createRedisClient` that is always creating a new client
- `getRedisClient` that is return the memoized one if exist or call createRedisClient if not